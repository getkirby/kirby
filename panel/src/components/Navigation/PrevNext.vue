<template>
  <k-button-group class="k-prev-next">
    <k-button v-bind="prev" icon="angle-left" />
    <k-button v-bind="next" icon="angle-right" />
  </k-button-group>
</template>

<script>
export default {
  props: {
    prev: {
      type: Object,
      default() {
        return {
          disabled: true,
          link: "#",
        }
      }
    },
    next: {
      type: Object,
      default() {
        return {
          disabled: true,
          link: "#"
        }
      }
    }
  },
};
</script>

<style lang="scss">
.k-prev-next {
  direction: ltr;
}
</style>
