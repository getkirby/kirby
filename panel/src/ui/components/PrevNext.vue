<template>
  <k-button-group class="k-prev-next">
    <k-button
      v-bind="prev ? prev : { disabled: true, link: '#' }"
      :icon="prevIcon"
    />
    <k-button
      v-bind="next ? next : { disabled: true, link: '#' }"
      :icon="nextIcon"
    />
  </k-button-group>
</template>

<script>
export default {
  props: {
    /**
     * Define whether the arrows point left-right or up-down.
     * Available options: `horizontal`|`vertical`
     */
    direction: {
      type: String,
      default: "horizontal"
    },
    prev: {
      type: [Boolean, Object],
      default: false
    },
    next: {
      type: [Boolean, Object],
      default: false
    }
  },
  computed: {
    prevIcon() {
      if (this.direction === "vertical") {
        return "angle-up";
      }

      return "angle-left";
    },
    nextIcon() {
      if (this.direction === "vertical") {
        return "angle-down";
      }

      return "angle-right";
    }
  }
};
</script>

<style lang="scss">
.k-prev-next {
  direction: ltr;
}
</style>
