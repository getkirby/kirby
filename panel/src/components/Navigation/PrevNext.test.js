import { describe, expect, it } from "vitest";
import { mount } from "@vue/test-utils";
import PrevNext from "./PrevNext.vue";

const setup = {
	stubs: ["k-button-group"]
};

describe.concurrent("PrevNext.vue", () => {
	it("has buttons", async () => {
		const wrapper = mount(PrevNext, {
			propsData: {
				prev: {
					link: "https://getkirby.com"
				}
			},
			...setup
		});

		expect(wrapper.vm.buttons).toEqual([
			{ link: "https://getkirby.com", icon: "angle-left" },
			{ disabled: true, link: "#", icon: "angle-right" }
		]);
		expect(wrapper.findComponent("k-button-group-stub").exists()).toBe(true);

		// see: https://github.com/getkirby/kirby/pull/4334
		//expect(wrapper.element).toMatchSnapshot();
	});

	it("has `button()` method", async () => {
		const wrapper = mount(PrevNext, setup);

		const defaults = {
			disabled: true,
			link: "#"
		};

		const config = {
			link: "https://getkirby.com"
		};

		expect(wrapper.vm.button()).toEqual(defaults);
		expect(wrapper.vm.button()).toEqual(defaults);
		expect(wrapper.vm.button(config)).toEqual(config);
	});

	it("has CSS selector", async () => {
		const wrapper = mount(PrevNext, setup);
		expect(wrapper.classes()).toContain("k-prev-next");
	});
});
