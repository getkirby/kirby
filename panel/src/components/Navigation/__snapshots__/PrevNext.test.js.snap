// Vitest Snapshot v1

exports[`PrevNext.vue > has CSS selector 1`] = `
<k-button-group-stub
  buttons="[object Object],[object Object]"
  class="k-prev-next"
/>
`;
